tensornetwork/network_components.py

```diff
@@ -554,7 +554,7 @@ def __init__(self,
       #always use the `Node`'s backend
       backend = tensor.backend
       tensor = tensor.tensor
-    if not backend:
+    if backend is not None:
```

Comment:
whoops, I believe you mixed up? it should be `if backend is None` instead of `if backend is not None`

Reply:
Yeah, Going through it i thought the check should be for `None`. The Issue mentioned `not None` to be added so went ahead with it. I will make that change.